plugin.audio.soundcloud/resources/plugin.py

```diff
@@ -0,0 +1,146 @@
+from future import standard_library
+standard_library.install_aliases()  # noqa: E402
+
+from resources.lib.soundcloud.api_v2 import ApiV2
+from resources.lib.kodi.cache import Cache
+from resources.lib.kodi.items import Items
+from resources.lib.kodi.search_history import SearchHistory
+from resources.lib.kodi.settings import Settings
+from resources.lib.kodi.vfs import VFS
+from resources.routes import *
+import urllib.parse
+import sys
+import xbmc
+import xbmcaddon
+import xbmcgui
+import xbmcplugin
+
+addon = xbmcaddon.Addon()
+addon_id = addon.getAddonInfo("id")
+addon_base = "plugin://" + addon_id
+addon_profile_path = xbmc.translatePath(addon.getAddonInfo('profile')).decode('utf-8')
+addon_temp_path = xbmc.translatePath('special://temp/%s' % addon_id).decode('utf-8')
```

Comment:
According to the guidelines, there has to be a good reason for not writing data to the userdata directory.

I saw that the twitch plugin also writes cache data to the temp folder, so I'm also using the temp dir for storing the cache data.

Reply:
As per the add-on rules wiki entry:
> add-ons should store all their data in their own subfolder inside the addon_data directory.

So if you could change that to the profile dir, that would be great :)
`xbmc.translatePath(addon.getAddonInfo('profile'))` (Not sure if `translatePath` is required)

Reply:
Yes, please cache on the userdata folder and manage it in the addon. /temp data won't have a reference to the running profile

Reply:
I fixed it. The plugin now uses a `cache` subdirectory inside the profile folder:
```